autosubmit/git/autosubmit_git.py

```diff
@@ -301,3 +301,16 @@ def check_unpushed_changes(expid: str) -> None:
                         # M: Modified, A: Added, D: Deleted, ?: Untracked
                         raise AutosubmitCritical("Push local changes to remote repository before running", 7075)
 
+    @staticmethod
+    def check_directory_in_use(dir_path: str) -> bool:
+        """
+        Lists all Open Files in a directory
+        """
+        result = subprocess.run(
+            ['lsof', '+D', dir_path],
```

Comment:
Lsof may not be available (e.g. docker containers).

We might have to either add it to the container recipe and installation instructions, or maybe use a python lib (not sure if psutil has something like lsof).

Reply:
`autosubmit stop` has code that checks if the experiment is running. Maybe that same function can be used to check if the expid is found in a process with `autosubmit run`, and if so warn the user.

Reply:
https://github.com/BSC-ES/autosubmit/blob/438743344dac7c910a3ad89758bab27abcf33829/autosubmit/helpers/processes.py#L162